lineapy/graph_reader/artifact_collection.py

```diff
@@ -101,34 +110,6 @@ def __init__(
                 )
                 raise KeyError(msg)
 
-    def _get_artifacts_grouped_by_session(
-        self, artifact_entries: List[LineaArtifactDef]
-    ) -> Dict[LineaID, List[LineaArtifact]]:
-        """
-        Get LineaArtifact from each artifact entry and group by the Session they belong to.
-
-        Artifact entries are specified as name and optionally version as the end user would specify.
-
-        This helper function is used to group target and reuse_precomputed artifacts so that we can
-        create SessionArtifacts for each Session.
-        """
-        artifacts_grouped_by_session: Dict[
-            LineaID, List[LineaArtifact]
-        ] = defaultdict(list)
-        seen_artifact_names: Set[str] = set()
-        for art_def in artifact_entries:
-            # Check no two target artifacts have the same name
-            if art_def["artifact_name"] in seen_artifact_names:
-                raise KeyError(
-                    "Artifact %s is duplicated", art_def["artifact_name"]
-                )
-            # Retrieve artifact and put it in the right group
-            art = LineaArtifact.get_artifact_from_def(self.db, art_def)
-            artifacts_grouped_by_session[art._session_id].append(art)
-            seen_artifact_names.add(art_def["artifact_name"])
-
-        return artifacts_grouped_by_session
-
```

Comment:
Based on the name alone, I would expect it only to do grouping, however, this seems to do three things:
 1) Check if the artifact name has been repeated in the input list
 2) converts artifactdef to a linea artfact by fetch it from the db
 3) groups these initialized artifacts into a dict with session ids as key

I think the function is doing too much and can be fragmented into three with each handling these tasks.